lib/formalist/child_forms/params_processor.rb

```diff
@@ -0,0 +1,34 @@
+require "json"
+require_relative "builder"
+
+module Formalist
+  module ChildForms
+    class ParamsProcessor
+      attr_reader :embedded_forms
+
+      def initialize(embedded_form_collection)
+        @embedded_forms = embedded_form_collection
+      end
+
+      def call(input)
+        return input if input.nil?
+
+        input = input.is_a?(String) ? JSON.parse(input) : input
+
+        input.inject([]) { |output, node| output.push(process(node)) }
+      end
+      alias_method :[], :call
+
+      private
+
+      def process(node)
+        node = node.dup
```

Comment:
Do we actually need to `dup` the node here? Seems like we're not mutating it at all, instead returning a new copy via the `.merge` line at the end of the method.